bibblio.py

```diff
@@ -0,0 +1,377 @@
+import json
+import logging
+import os
+import re
+from datetime import datetime, timedelta
+from nose.tools import set_trace
+from urlparse import urlparse
+
+from bs4 import BeautifulSoup
+from sqlalchemy.orm import eagerload
+
+from core.coverage import (
+    CoverageFailure,
+    CoverageProvider,
+    CoverageRecord,
+)
+from core.model import (
+    Credential,
+    CustomList,
+    CustomListEntry,
+    DataSource,
+    Edition,
+    Hyperlink,
+    Identifier,
+    Representation,
+    Resource,
+    Work,
+)
+from core.util.epub import EpubAccessor
+from core.util.http import HTTP
+
+from config import Configuration
+
+
+class BibblioAPI(object):
+
+    API_ENDPOINT = u'https://api.bibblio.org/v1/'
+    CATALOGUES_ENDPOINT = API_ENDPOINT + u'catalogues/'
+    CONTENT_ITEMS_ENDPOINT = API_ENDPOINT + u'content-items/'
+
+    TOKEN_CONTENT_TYPE = u'application/x-www-form-urlencoded'
+
+    log = logging.getLogger(__name__)
+
+    @classmethod
+    def from_config(cls, _db):
+        config = Configuration.integration(Configuration.BIBBLIO_INTEGRATION)
+        if not (config and len(config.values())==2):
+            return None
+
+        client_id = config.get(Configuration.BIBBLIO_ID)
+        client_secret = config.get(Configuration.BIBBLIO_SECRET)
+        return cls(_db, client_id, client_secret)
```

Comment:
I'd really like to start using `ExternalIntegration` to store this stuff instead of the JSON config, but since this isn't a `Collection` I don't see a good way to figure out which `ExternalIntegration` has the Bibblio configuration. @aslagle, do you have any ideas about this part? Maybe we need to add a `ExternalIntegration.name` so it's possible to look up an integration by some specific key?